G/gRPC/build_tarballs.jl

```diff
@@ -0,0 +1,71 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "gRPC"
+version = v"1.48.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/grpc/grpc.git", "d2054ec6c6e8abcecf0e24b0b4ee75035d80c3cc")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/grpc && \
+git submodule update --init --recursive && \
+mkdir -p cmake/build && \
+pushd "cmake/build" && \
+cmake \
+    -DgRPC_INSTALL=ON \
+    -DgRPC_BUILD_TESTS=OFF \
+    -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DgRPC_SSL_PROVIDER=package \
+    ../.. && \
+make && \
+make install && \
+popd && \
+mkdir -p cmake/build && \
+pushd "cmake/build" && \
+cmake \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DgRPC_BUILD_TESTS=OFF \
+    -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    ../.. && \
+make && \
+make install && \
+popd
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform(
+      "x86_64", "linux";
+      libc="glibc",
+      libgfortran_version=v"5.0.0",
+      libstdcxx_version = v"3.4.29",
+      cxxstring_abi = "cxx11"
+    )
+]
+platforms = expand_gfortran_versions(platforms)
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("grpc_cpp_plugin", :grpc_cpp_plugin),
+    ExecutableProduct("grpc_csharp_plugin", :grpc_csharp_plugin),
+    ExecutableProduct("grpc_objective_c_plugin", :grpc_objective_c_plugin),
+    ExecutableProduct("grpc_python_plugin", :grpc_python_plugin),
+    ExecutableProduct("grpc_ruby_plugin", :grpc_ruby_plugin),
+    ExecutableProduct("grpc_node_plugin", :grpc_node_plugin),
+    ExecutableProduct("grpc_php_plugin", :grpc_php_plugin)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("OpenSSL_jll")
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; lock_microarchitecture=false)
```

Comment:
Why `lock_microarchitecture=false`? :weary: Also, going forward we want to require julia v1.6, with `julia_compat="1.6"`.

Reply:
I'm not sure why that's there, probably a copypaste leftover. I'm removing it and testing locally right now.